bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PageListItemImpl.java

```diff
@@ -107,4 +108,7 @@ private Page getRedirectTarget(@NotNull Page page) {
         return result;
     }
 
+    private boolean isShadowingEnabled(@NotNull SlingHttpServletRequest request) {
```

Comment:
I see you defined both a policy config and a dialog property for disabling shadowing, but you're only using the latter. What's the purpose of defining the policy config property? Am I missing something?

Also, I would prefer you inject this property at component (navigation/breadcrumb) level and pass it on to this logic.

When navigating such a long path of methods it might be safer to null-check.

Reply:
Hi @vladbailescu, thanks for a fast code review :) 

Good point, I will add a fallback to currentStyle like it's done for rest of properties: 
`isShadowingEnabled = !properties.get(PN_DISABLE_SHADOWING, currentStyle.get(PN_DISABLE_SHADOWING, false));`

  I was planning to put the property on Navigation component level initially, but then thought it might be also helpful for other components, that use PageListItemImpl (e.g. List).  Now when I look at it a second time, I see it's more clear way to keep it on a component level. I’ll move it.